### PR TITLE
Refactor

### LINKED ISSUE #9: Node v4
The Node Foundation [will stop maintenance on Node versions < 4](https://github.com/nodejs/LTS#lts-schedule) after December.

I'd be happy to refactor the module so it uses Promises and constants internally. This would benefit the readability of the code.

Are you interested in this? If so, I'll create a PR with my suggestions.

### LINKED ISSUE #14: Promise support
Any plan to add support for promises?

// @hemanth

### PR DESCRIPTION
**Pull request by Richienb**
This PR refactors the codebase, making extensive use of promises and modern code practices. I've also migrated jslint, jshint and mocha to xo and ava configured similarly and I've added some typings.

If you'd like me to continue maintaining this repo in the future, please add me as a collaborator.

___

Fixes #11, Fixes: #14, Fixes: #9 

// @hemanth

### CODE REVIEW

**Comment by hemanth** [comment]
@Richienb Rad! Thanks for re-write.

What is your npm handle? [Will add you as one of the owners] and then you can release `v2` thanks!

File: index.js [hunk 1]
```diff
@@ -1,136 +1,27 @@
...
-    throw new TypeError('Bad arguments');
+const fs = require('fs');
+const {promisify} = require('util');
+const tempWrite = require('temp-write');
+const pathExists = require('path-exists');
+const pipeline = promisify(require('stream').pipeline);
+
+module.exports = async (filename, data) => {
+  if (await pathExists(filename)) {
+    const temporaryFile = await tempWrite(data);
```
Comment (papb) [inline index.js:10]: Hi @Richienb, why is it necessary to write to a temp file first?

**Comment by Richienb** [comment]
@hemanth My npm handle is `richienb`.

File: index.js (hunk 1 above)
Comment (Richienb) [reply index.js:10]: https://stackoverflow.com/a/15424018/8384910

Comment (papb) [reply index.js:10]: Hmm I see. I think I have an optimization in mind though. Might open a PR in the future.

**Comment by hemanth** [comment]
```
$ npm owner add richienb prepend-file
+ richienb (prepend-file)
```

**Comment by Richienb** [comment]
@hemanth Please also add me as a collaborator for the repo so I can create release tags.

**Comment by Richienb** [comment]
// @hemanth

**Comment by hemanth** [comment]
[Done](https://github.com/hemanth/node-prepend-file/invitations)

**Pull request merged**